### PR TITLE
 Correct H50get_info deprecation for earlier releases

### PR DESCRIPTION
**Pull request by musm**
close https://github.com/JuliaIO/HDF5.jl/issues/508

by retaining old function behavior in H50get_info

### CODE REVIEW

**Comment by musm** [comment]
merging sooner than usual since this is uncontroversial and causing headaches to users.

**Comment by BoundaryValueProblems** [comment]
@musm , unfortunately, I won't be able to test this latest version since REQUIRE file specifies it as Julia v0.7. Many of us, due to the problem of Homebrew under Julia v0.7, the common situation, at least on macOS is the following:
* Julia v0.6.4 + HDF5.jl v0.9.4 + HDF5 library 1.10.3 (if Homebrew.update() was run);
* Julia v0.7/1.0 + HDF5.jl v0.10.1 + HDF5 library 1.10.2 (since Homebrew.update() does not work).

In other words, I could test HDF5.jl v0.10.1 + HDF5 library 1.10.2 combination, and Pkg.test passed. However, I still cannot test the combination of HDF5.jl v0.10.1 + HDF5 library 1.10.3 at this point due to the problem of Homebrew.jl under Julia v0.7/1.0.
Thanks!

**Comment by BoundaryValueProblems** [comment]
@musm Thanks a lot! I just upgraded to HDF5.jl v0.9.5, and ran the Pkg.test("HDF5") on my MacBook Pro with the following setup:
* Julia v0.6.4 + HDF5.jl v0.9.5 + HDF5 library 1.10.3

 And the tests passed!!
